testing/sdk_tasks.py

```diff
@@ -423,14 +423,20 @@ def _check_tasks_updated():
     _check_tasks_updated()
 
 
-def check_tasks_not_updated(service_name, prefix, old_task_ids):
+def check_tasks_not_updated(
```

Comment:
Did some digging around and I think I know what's happening:
1. I'd guess that the V1 API is faster to mark the framework as inactive, or maybe it's instead slower to mark the framework active again after the scheduler has re-registered?
2. sdk_tasks.py:L204 filters out frameworks with `!active` when fetching the list of all frameworks.
3. Given 1+2, `get_task_ids()` would return an empty list if the scheduler hasn't completed re-registering.

Given this, I think a better solution here would be:
- Implement a function (wait_for_active_framework()?) that waits for at least one framework with the specified service name to be active. This could be built using logic similar to the `/mesos/frameworks` fetch in `sdk_tasks._get_service_tasks()`.
- Update the hdfs test to invoke that new function before calling check_tasks_not_updated(), which shouldn't need a retry anymore
- Couldn't hurt to add a wait_for_active_framework()+check_tasks_not_updated() check to hello-world test_zzzrecovery.kill_scheduler too, so that this also gets checked there (it isn't currently)